### ISSUE TITLE
OTel Export is slow at high rps and no aggregation.

### ISSUE THREAD

**Issue by philkuz** [pixie-io/pixie#956]
**Describe the bug**
OTel export seems very slow when hit with reasonable load. This causes the queries to get stuck inside of the carnot queue.
The export code is not very optimized so there is a bunch of opportunities for improvement.

We haven't ran into this before because we typically aggregate metrics, but we should be aware that this is an upper-bound we can move.
**To Reproduce**
On a cluster w/ 200rps, the following will cause problems
```
import px
df = px.DataFrame(table='http_events', start_time="-30s")
df.service = df.ctx['service']

px.export(
  df, px.otel.Data(
  endpoint=px.otel.Endpoint(
    url='<replace-with-url>',
    insecure=True
  ),
    resource={
      # User must specify service_name
      'service.name': df.service,
    },
    data=[
      px.otel.metric.Gauge(
        name='http_events.req_body_size',
        description='Request body size (before any truncation)',
        value=df.req_body_size,
        attributes={
        },
      ),
    ],
  ),
)
```
**Expected behavior**
This should be able to serialize fast enough.

**Additional context**
The place to optimize will be here: `src/carnot/exec/otel_export_sink_node.cc`

**Comment by JamesMBartlett** [comment]
I tried this on a cluster with 3000rps of http events, using a cluster local otel collector, and the script completed in 100ms. So I think a lot of the slowness depends on the otel collector used.

**Comment by caphrim007** [comment]
@JamesMBartlett which otel collector did you test with and can you provide the running config?

**Comment by caphrim007** [comment]
some more context on this after a long day of debugging.

the telegraf configuration we have i place includes the usage of this plugin here

https://github.com/influxdata/telegraf/tree/release-1.25/plugins/processors/aws/ec2

This processor has a limitation insofar as it does not cache (mostly) immutable data that it receives from calls it makes to the AWS IMDS service. The result of this is that its default parallelism (10) is insufficient to prevent telegraf from bottlenecking, but changing that limit to a higher value results in rate-limiting by AWS and losing valuable tag data.

The gist is that when we removed this plugin from our config, we saw the 30s latency drop back to 300ms.

We've since opened up a request with telegraf to support caching (even if only optionally) within this processor, and in the meantime we forked the processor to add the caching as we specifically needed it.
